index.d.ts

```diff
@@ -0,0 +1,40 @@
+declare module 'envvar' {
+  /**
+   * Retrieves a given environment variable as a string.
+   *
+   * @throws if the environment variable is not set and a default is not
+   *         specified.
+   */
+  export function string(name: string, defaultValue?: string): string;
+
+  /**
+   * Retrieves a given environment variable as a number.
+   *
+   * @throws if the environment variable is not set and a default is not
+   *         specified.
+   * @throws if the value of the environment variable does not represent a
+   *         number.
+   */
+  export function number(name: string, defaultValue?: number): number;
+
+  /**
+   * Retrieves a given environment variable as a boolean.
+   *
+   * @throws if the environment variable is not set and a default is not
+   *         specified.
+   * @throws if the value of the environment variable does not represent a
+   *         boolean.
+   */
+  export function boolean(name: string, defaultValue?: boolean): boolean;
+
+  /**
+   * Retrieves a given environment variable, validating that its value is
+   * contained within the specified enum.
```

Comment:
I find the use of _enum_ slightly confusing now that we're no longer using the word elsewhere. How about the following?

_Returns the value of the specified environment variable, provided that the value is one of the permissible strings._